### PR TITLE
tracer: implement zipkin getTraceIdAsHex

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: make zipkin tracer work with opentelemetry accesslogger
Additional Description: follow up https://github.com/envoyproxy/envoy/pull/33281
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/access_loggers/open_telemetry/access_log_impl.cc [hunk 1]
```diff
@@ -90,8 +90,15 @@ void AccessLog::emitLog(const Formatter::HttpFormatterContext& log_context,
...
-      absl::HexStringToBytes(log_context.activeSpan().getTraceIdAsHex());
+
+  // Setting the trace id if available.
+  std::string trace_id_hex = log_context.activeSpan().getTraceIdAsHex();
+  if (trace_id_hex.size() == 32) {
+    *log_entry.mutable_trace_id() = absl::HexStringToBytes(trace_id_hex);
+  } else if (trace_id_hex.size() == 16) {
+    auto trace_id = absl::StrCat(Hex::uint64ToHex(0), trace_id_hex);
+    *log_entry.mutable_trace_id() = absl::HexStringToBytes(trace_id);
+  }
```
Comment (wbpcode) [inline source/extensions/access_loggers/open_telemetry/access_log_impl.cc:101]: Could you add some comments to explain these logic? Thanks.

/wait

Comment (zirain) [reply source/extensions/access_loggers/open_telemetry/access_log_impl.cc]: update

**Review by wbpcode — APPROVED** [review]

**Comment by wbpcode** [comment]
Will merge this after the latest 1.30 is released (today or tomorrow). Feel free to ping me is I forget it.

/wait

**Comment by zirain** [comment]
kindly ping @wbpcode since 1.30 is released.

**Pull request merged**